web/src/main/java/org/fao/geonet/monitor/link/LinkMonitorService.java

```diff
@@ -0,0 +1,142 @@
+import org.fao.geonet.GeonetContext;
+import org.fao.geonet.constants.Geonet;
+import org.fao.geonet.kernel.search.IndexAndTaxonomy;
+import org.fao.geonet.kernel.search.index.GeonetworkMultiReader;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LinkMonitorService extends LinkMonitorInterface {
+
+    public enum Status {
+        FAILED,
+        WORKING,
+        UNKNOWN
+    }
+
+    public static int percentWorking = 90;
+
+    public static int maxChecks = 10;
+
+    public static int timeout = 15;
+
+    public static int freshness = 60 * 60;
+
+    private boolean unknownAsWorking = true;
+
+    // Milliseconds between running checks on every record. This is here to
+    // prevent undesired hammering of servers
+    private static int betweenChecksIntervalMs = 100;
+
+    private final Map<String, MetadataRecordInfo> recordMap;
+
+    private long reindexTimestamp = -1;
+
+    public LinkMonitorService() {
+        this.recordMap = new HashMap<String, MetadataRecordInfo>();
+    }
+
+    @Override
+    public void init(ResourceManager resourceManager, GeonetContext geonetContext, Logger logger, long reindexInterval) {
+        super.init(resourceManager, geonetContext, logger, reindexInterval);
+    }
+
+    private ArrayList<String> getAllRecords() {
+        try {
+            return getAllRecordsLucene();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private ArrayList<String> getAllRecordsLucene() throws Exception {
+        ArrayList<String> records = new ArrayList<String>();
+
+        IndexAndTaxonomy indexAndTaxonomy = getGeonetContext().getSearchmanager().getNewIndexReader(null);
+        try {
+            GeonetworkMultiReader reader = indexAndTaxonomy.indexReader;
+
+            int capacity = (int)(reader.maxDoc() / 0.75) + 1;
+            for (int i = 0; i < reader.maxDoc(); i++) {
+                DocumentStoredFieldVisitor idChangeDateSelector = new DocumentStoredFieldVisitor("_uuid", "_changeDate");
+                reader.document(i, idChangeDateSelector);
+                org.apache.lucene.document.Document doc = idChangeDateSelector.getDocument();
+                String uuid = doc.get("_uuid");
+
+                if (uuid == null) {
+                    Log.error(Geonet.INDEX_ENGINE, "Document with no _uuid field skipped! Document is " + doc);
+                    continue;
+                } else {
+                    getLogger().debug(String.format("Link Monitor Service adding uuid '%s'", uuid));
+                    records.add(uuid);
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            getGeonetContext().getSearchmanager().releaseIndexReader(indexAndTaxonomy);
+        }
+
+        return records;
+    }
+
+    private boolean needReindex() {
+        long now = System.currentTimeMillis() / 1000l;
```

Comment:
Seems to be a lot of conversion to seconds going on all over the place - why not just deal in ms?  That's typical (although not *always* the case) when dealing in timestamps as longs, anyway.